src/Identity/Core/src/IdentityApiEndpointRouteBuilderExtensions.cs

```diff
@@ -42,44 +59,64 @@ public static IEndpointConventionBuilder MapIdentityApi<TUser>(this IEndpointRou
         {
             var userManager = sp.GetRequiredService<UserManager<TUser>>();
 
+            if (!userManager.SupportsUserEmail)
+            {
+                throw new NotSupportedException($"{nameof(MapIdentityApi)} requires a user store with email support.");
+            }
+
+            var userStore = sp.GetRequiredService<IUserStore<TUser>>();
+            var emailStore = (IUserEmailStore<TUser>)userStore;
+
             var user = new TUser();
-            await userManager.SetUserNameAsync(user, registration.Username);
+            await userStore.SetUserNameAsync(user, registration.Username, CancellationToken.None);
+            await emailStore.SetEmailAsync(user, registration.Email, CancellationToken.None);
             var result = await userManager.CreateAsync(user, registration.Password);
 
             if (result.Succeeded)
             {
+                await SendConfirmationEmailAsync(user, userManager, registration.Email);
                 return TypedResults.Ok();
             }
 
-            return TypedResults.ValidationProblem(result.Errors.ToDictionary(e => e.Code, e => new[] { e.Description }));
+            return CreateValidationProblem(result);
         });
 
-        routeGroup.MapPost("/login", async Task<Results<UnauthorizedHttpResult, Ok<AccessTokenResponse>, SignInHttpResult>>
-            ([FromBody] LoginRequest login, [FromQuery] bool? cookieMode, [FromServices] IServiceProvider sp) =>
+        routeGroup.MapPost("/login", async Task<Results<Ok<AccessTokenResponse>, ProblemHttpResult, IResult>>
+            ([FromBody] LoginRequest login, [FromQuery] bool? cookieMode, [FromQuery] bool? persistCookies, [FromServices] IServiceProvider sp) =>
         {
-            var userManager = sp.GetRequiredService<UserManager<TUser>>();
-            var user = await userManager.FindByNameAsync(login.Username);
+            var signInManager = sp.GetRequiredService<SignInManager<TUser>>();
+
+            signInManager.PrimaryAuthenticationScheme = cookieMode == true ? IdentityConstants.ApplicationScheme : IdentityConstants.BearerScheme;
+            var isPersistent = persistCookies ?? true;
```

Comment:
Do we care if `persistCookies` is set when `cookieMode` is `false` or `null`?

Reply:
It doesn't make a difference.